qt/scientific_interfaces/Indirect/ConvFitDataPresenter.cpp

```diff
@@ -24,7 +24,7 @@ namespace CustomInterfaces {
 namespace IDA {
 
 ConvFitDataPresenter::ConvFitDataPresenter(ConvFitModel *model,
-                                           IIndirectFitDataView *view)
+                                           IndirectFitDataView *view)
```

Comment:
Can I ask why the interfaces have been removed? Are they not required for mocking?

Reply:
They were remnants of a prevous refactor, since then they had become redundant and could be removes as everything they had was inhereted by onle a single class that overwrote everything.

Reply:
Thanks for clarifying that. Is it that the interface classes like `IIndirectFitDataView` were intended to be there so gmock could be used to put mocks in place rather than the real thing? This is quite common for us where we have the real implementation and then a mock in the unit tests but C++ requires the presence of a base interface class. How will the testing be achieved without the presence of the interface?

Reply:
I had not realised that might be a reason for them to be there, all the current tests are working by replacing the base class in the mock in tests. e.g.
https://github.com/mantidproject/mantid/blob/f5889bb5d12853ae8c5d49b7a1e9ae313ac10391/qt/scientific_interfaces/Indirect/test/IndirectFitDataPresenterTest.h#L59

Reply:
Oh I see. That will work but it does require the base class to be constructed too so more traditionally you would have the interface, here `IIndirectFitDataView`, be completely empty so that there are no uninteneded side effects from the mock. Having an interface also means replacing the implementation with a different class that just provides the methods from the  pure virtual class is easier and it isn't required to use the existing implementation at all.

Reply:
In that case I suppose it would be a good idea to put them back.

Reply:
I think for long-term flexibility it would be better to depend on [interfaces rather than implementations](https://en.wikipedia.org/wiki/Dependency_inversion_principle#Dependency_inversion_pattern). Thanks for looking back at this.